webdriver-spec.html

```diff
@@ -1435,288 +1435,513 @@
 
-     <li><p>Let <var>socks proxy port</var> be the result of <a>getting a property</a>
-      named <code>socksProxyPort</code> from <var>proxy</var>.
+  <dt>"<code>acceptSslCerts</code>"
+  <dd>False, indicating the session <em>will not</em> implicitly trust untrusted
+  or self-signed SSL certificates on <a data-lt=go>navigation</a>.
+  </dl>
 
-     <li><p>If <var>socks proxy</var> and <var>socks proxy port</var> is defined:
+<li><p>Let <var>capabilities length</var> be the number of entries in <var>capabilities</var>.
 
-      <ol>
-        <li><p>Let <var>socks version</var> be the result of <a>getting a property</a>
-         named <code>socksVersion</code> from <var>proxy</var>.
+<li><p>Let <var>k</var> be 0.
 
-        <li><p>Let <var>socks username</var> be the result of <a>getting a property</a>
-         named <code>socksUsername</code> from <var>proxy</var>.
+<li><p>While <var>k</var> &lt; <var>capabilities length</var>:
 
-        <li><p>Let <var>socks password</var> be the result of <a>getting a property</a>
-         named <code>socksPassword</code> from <var>proxy</var>.
-      </ol>
+  <ol>
+  <li><p>Let <var>capability name</var> be the name of the entry at index
+  <var>k</var> in <var>capabilities</var>.
 
-     <li><p>Follow implementation defined steps to set the proxy
-      using defined variables from previous steps.
-      If there are items that can not be set during this process return <a>error</a>.
-    </ol>
+  <li><p>Let <var>capability value</var> be the value of the entry at index
+  <var>k</var> in <var>capabilities</var>.
 
-   <dt>Otherwise
-   <dd><p>Return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
-  </dl>
+  <li><p>Switch on <var>capability name</var>:
+
+    <dl class=switch>
+      <dt>"<code>browserName</code>"
+      <dd><p>If <var>capability value</var> is not equal to
+      "<code>browserName</code>" entry in <var>matched capabilities</var>,
+      return null.
+
+      <dt>"<code>browserVersion</code>"
+      <dd><p>Compare <var>capability value</var> to the
+      "<code>browserVersion</code>" entry in <var>matched capabilities</var>
+      using an implementation defined comparison algorithm. The comparison
+      <em class="rfc2119" title="SHOULD">SHOULD</em> accept a
+      <var>capability value</var> that places constraints on the version using
+      the "<code>&lt;</code>", "<code>&lt;=</code>", "<code>&gt;</code>", and
+      "<code>&gt;=</code>" operators.
+
+      <p>If the two values do not match, return null.
+
+      <p class=note>Version comparison is left as an implementation detail since
+      each user agent will likely have conflicting methods of encoding the
+      user agent version and standardizing these schemes is beyond the scope of
+      this specification.
+
+      <dt>"<code>platformName</code>"
+      <dd><p>If <var>capability value</var> is not equal to
+      "<code>platformName</code>" entry in <var>matched capabilities</var>,
```

Comment:
We should canonicalise the common platform names (Windows, Linux, and OS X), otherwise there's less hope of interop between competing services implementing this spec. That is CompanyX could define a linux node as "linux", CompanyY as "gnu/linux", and then someone wishing to switch services would need to recode their tests. We should also indicate the name of the property used for determining the desired platform.